kube-client/src/config/mod.rs

```
@@ -299,6 +295,8 @@ impl Config {
         let cluster_url = loader
             .cluster
             .server
+            .clone()
+            .unwrap_or_default()
```

This should return an error for missing cluster url.

---

Done